### PR TITLE
[nlcglib] New package recipe

### PR DESCRIPTION
**Pull request by abussy**
This is a new build recipe for the [nlcglib](https://github.com/simonpintarelli/nlcglib) library, providing a set of nonlinear CG (conjugated gradient) methods for wave-function minimization in DFT (density functional theory). We made the conscious decision not to support Windows or FreeBSD.

### CODE REVIEW

File: N/nlcglib/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  CMAKE_ARGS="${CMAKE_ARGS} \
+               -DMPI_C_LIB_NAMES='mpi;pmpi;hwloc' \
+               -DMPI_CXX_LIB_NAMES='mpicxx;mpi;pmpi;hwloc' \
+               -DMPI_mpicxx_LIBRARY=${libdir}/mpich/lib.libmpicxx.a \
+               -DMPI_mpi_LIBRARY=${libdir}/mpich/lib/libmpi.a \
+               -DMPI_pmpi_LIBRARY=${libdir}/mpich/lib/libpmpi.a \
+               -DMPI_hwloc_LIBRARY=${libdir}/libhwloc.dylib"
+fi
+
+cmake .. ${CMAKE_ARGS} || cmake .. ${CMAKE_ARGS}
```
Comment (imciner2) [inline N/nlcglib/build_tarballs.jl:49]: Why are you running CMake twice?

Comment (abussy) [reply N/nlcglib/build_tarballs.jl]: I am not sure how it ended up there, it works fine with a single CMake call... Thanks for pointing it out, I cleaned it up.

**Comment by imciner2** [comment]
I don't have much experience with MPI-based recipes here, so @giordano @eschnett either of you able to take a look at the MPI parts?

File: N/nlcglib/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+products = [
+   LibraryProduct("libnlcglib", :libnlcglib)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("OpenBLAS32_jll"),
+    Dependency("Kokkos_jll"),
+    Dependency("nlohmann_json_jll"),
```
Comment (giordano) [inline N/nlcglib/build_tarballs.jl:76]: You said this is header-only, right?  See https://docs.binarybuilder.org/stable/build_tips/#Dependencies-for-the-target-system-vs-host-system
```suggestion
    BuildDependency("nlohmann_json_jll"),
```

**Pull request merged**